### ISSUE TITLE
Awaitility untilAsserted does not work

### ISSUE THREAD

**Issue by kevinfeyrer** [SpectoLabs/hoverfly-java#241]
## Description of the bug
The Hoverfly Java implementation of verify does not work nicely with Awaitility.

This happens because `io.specto.hoverfly.junit.verification.HoverflyVerificationError` extends `java.lang.Error` and Awaitility will only continue polling if the exception extends `java.lang.AssertionError`.

## Steps to reproduce the issue
```Java
private void delayedRequest() {
    new Thread(() -> {
        try {
            Thread.sleep(500);
            String host = String.format("localhost:%s", hoverfly.getHoverflyConfig().getProxyPort());
            HttpHost proxy = HttpHost.create(host);
            HttpClientBuilder.create()
                .setProxy(proxy)
                .build()
                .execute(new HttpPost("http://service/path"));
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }
    }).start();
}

@Test
public void aTest() {
  delayedRequest();

  await().pollInterval(Duration.ofMillis(100))
    .untilAsserted(() -> {
      hoverfly.verify(service("service")
          .post("/path")
          .anyBody(),
        times(1));
    });
}
```

### Observed result
Failed unit test.

### Expected result
Passed unit test.

## Additional relevant information
Java 11
Hoverfly 0.13.0
Awaitility 4.0.3

**Comment by tommysitu** [comment]
Hey Kevin, thanks for pointing this out 👍  I've got a fix ^^ which incorporated your test.

**Comment by tommysitu** [comment]
fixed in v0.14.0
